text/0000-create-stackify-repo.md

```diff
@@ -0,0 +1,159 @@
+# Meta
+[meta]: #meta
+- Name: Create stackify repo
+- Start Date: 2020-11-06
+- Authors: @martyspiewak @mdelillo @dumez-k
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+Create a new repo called “stackify” which can be used to turn OCI images into CNB compliant stack images.
+
+# Definitions
+[definitions]: #definitions
+N/A
+
+# Motivation
+[motivation]: #motivation
+Currently, for users to extend the published stack images, or turn their own images into stack images, they have to create Dockerfiles which take the base images, apply their changes, and then set the CNB metadata. It can be difficult to generate and set all of the relevant metadata, especially the mixins.
+
+Stackify will be a tool that they can use to perform all of these manipulations to their base image without Dockerfiles.
+
+
+# What it is
+[what-it-is]: #what-it-is
+Stackify will be a tool that will be useful to users who would like to customize the published stack images or turn their own base images into CNB compliant stack images.
+
+Users can use stackify to add packages and ca-certificates to a base image as well as add all required CNB metadata to the image.
+
+Stackify can be integrated into Pack in the form of a `pack stack create` command.
+
+# How it Works
+[how-it-works]: #how-it-works
+For stacks defined by the buildpacks project, Stackify will add mixin labels for each package installed on the image. If Stackify is used to add a stack ID other than one defined by the buildpacks project, mixin labels will not be added.
+
+If Stackify is used with an already valid stack image, the `CNB_*` labels and users will not be changed unless new user and group ID's are passed, in which case the labels and user will be overwritten.
+
+If run on the image being "stackify-ed", Stackify can be used to add packages and ca-certificates.
+
+Stackify will use Kaniko to create additional image layers with the additional packages and ca-certificates that users choose to add to their base image. It will also use GGCR to manipulate the image metadata.
+
+
+### Proposed Interface
+Stackify will be run once for each image (build and run):
+```
+stackify 
+  --base-image string
+      Source image ref
+  --destination string
+      Destination for stack image
+  --stack-id string
+      Stack ID to add to image metadata
+  --certs-dir string
+      Path to directory with CA Certs to add to image
+  --mixin-labels string
+      Comma-separated list of mixins to add to labels
+  --packages string
+      Comma-separated list of packages to add to image
+  --user-id string
+      CNB User ID
+  --group-id string
+      CNB Group ID
```

Comment:
I am curious about how stackify would handle the cases when the specified user-id/group-id do not exist on the base image. Would it create them or would it throw an error?

Reply:
It would create them.

Reply:
Gotcha! I hit some errors in the past when I was trying to use extremely minimal base images as runtime images as they didn't have useradd or even /bin/sh. I'm hoping that this would be able to handle such situations by manually creating the required files.